### ISSUE TITLE
Moving window with Ctrl+W + L, toggle with wrong windows size

### ISSUE THREAD

**Issue by kiuKisas** [akinsho/toggleterm.nvim#51]
https://user-images.githubusercontent.com/12698940/122531483-d6214400-d01f-11eb-90d1-e138ad0ae5bc.mp4

Steps:
- open `ToggleTerm`
- change window position with `Ctrl-W` + `L`
- close term with `:ToggleTerm`
- open term with `:ToggleTerm`
--> wrong size (full width)

**Comment by akinsho** [comment]
@kiuKisas tbh this is not really a use case this plugin can support easily and one I personally am not particularly keen to add. I'll explain.

When you create a toggleterm you specify the orientation for the terminal and the size gets saved. When you then manually re-align it we don't try to figure out what orientation the window is now in and what it's new dimensions should be since I only save the size which for a horizontal split is the vertical height and for a vertical split is the width.

Basically keeping track of manually altered window properties is kind of another ball game and one I don't really want to get into tbh.

**Comment by rmagatti** [comment]
Hey @akinsho, great plugin! I had my own solution for this but yours is much more polished than what I had.

I believe what @kiuKisas suggested could indeed be achieved, but as you say, maybe not easily.

I imagine when toggling a specific terminal, one could at that point grab the window options and save them for a specific terminal number, then subsequently when reopening, setting those for the window to be opened.

You wouldn't then need to keep track of changes to the window when they happen, but only when the window is closed/toggled off.

One could also set an autocmd for, say, `WinLeave {toggleTermPattern}` when opening the window that will trigger the function that grabs the window's properties too instead of relying on explicit toggling with the mappings or commands the plugin provides.

Anyway, yeah, doable, but not easy 😅. If someone were to work on this, would you consider accepting a PR?

**Comment by akinsho** [comment]
Hi @rmagatti 👋🏾 ,

so regarding the window properties there was actually a PR from someone a while back to `persist_size` the idea being as you say on `WinLeave` to keep track of the window size. I refactored this plugin a while ago and I'm not sure how well that change survived. So basically there is already a bit of logic already trying to track window props, the difficulty here isn't *when* to get the window properties. As you suggest `WinLeave` will work the issue is *seeing the change to orientation* and recording it.

The main issue just to be clear is that currently `size` which is a prop users depend on means either height or width so when the window orientation changes and it's eventually re-opened we already call a function to resize the terminal to the expected size. The problem with this though is if the orientation has changed that works incorrectly.

Sorry that was a lot of pre-amble, regarding the fix and getting the window options on `WinLeave` unfortunately there isn't a very easy/dependable way to figure out if a window is vertical or horizontal AFAIK. There is no `vim.wo.splitdirection` or something. I think you could potentially use `vim.fn.winlayout` and try to derive the `direction` from that and then update it's `direction` property (and correct width or height as size) inside toggle term so the the size refers to the right axis.

Hopefully that makes sense, also a PR would be fantastic 🚀 🙌🏾 .

Regarding implementation I think it'd be:
* Find existing `WinLeave` persist logic (this logic is probably fragmented and buggy now since the refactor)
* Add extra logic to derive the current window orientation e.g. has it changed if so update the terminal's direction
* Update the size as well based on the correct/update orientation

**Comment by akinsho** [comment]
Going to close this since it's been quite a while. I'm still open to this idea, but *I personally* have no idea how this could actually be done, as mentioned earlier there is no api to easily tell me if a window is horizontal or vertical and any heuristic will be a bit buggy. If someone comes along who know *for sure,* please post here, but otherwise I'll close this out rather than have it open for years more.
